unit/util/range.cpp

```diff
@@ -188,6 +189,34 @@ SCENARIO("range tests", "[core][util][range]")
       REQUIRE(input2 == expected_result2);
     }
   }
+  GIVEN("A vectors of int and a list of strings.")
+  {
+    std::vector<int> int_vector{1, 2};
+    std::list<std::string> string_list{"foo", "bar"};
```

Comment:
It would be good to have an equivalent test where `int_vector` and `string_list` are defined const.